packages/next/client/link.tsx

```diff
@@ -169,6 +169,8 @@ class Link extends Component<LinkProps> {
       scroll = as.indexOf('#') < 0
     }
 
+    performance.mark('linkClick') // marking link clicks as a navigation start entry
```

Comment:
Should this actually be moved to the router `push` and `replace` method? Because otherwise you're missing out on the programmatic router usage.

Reply:
That might also be useful as you could track shallow vs not shallow

Reply:
Do you mean right after the promise succeeds? Like so:

```
    Router[this.props.replace ? 'replace' : 'push'](href, as, {
      shallow: this.props.shallow,
    })
      .then((success: boolean) => {
        if (!success) return

        performance.mark('linkClick')
      })
```

Tried this at first, but since it's async - this doesn't get picked up until `render` finishes firing (but please correct me if I'm wrong however)